mypy/types.py

```diff
@@ -896,8 +896,9 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, Instance):
             return NotImplemented
         return (self.type == other.type
-                and self.args == other.args
-                and self.last_known_value == other.last_known_value)
+                and self.args == other.args)
+        # TODO: last_known_value is not set correctly at initializations of Literal type
+        # and self.last_known_value == other.last_known_value)
```

Comment:
This change is not safe, since now two `Instance` objects that compare equal can have different hash values. 

Since these support hashing, it seems necessary to also use `last_known_value` when comparing for equality. Otherwise all sorts of weird (and hard-to-debug) things could happen.

I think that fixing the value of `last_known_value` could be the way to go. Alternatively, it may be feasible to replace `==` with comparing `type` and `args` attributes in selected contexts only (e.g. replace `i1 == i2` with `i1.type == i2.type and i1.args == i2.args`), but I'm not sure about this.